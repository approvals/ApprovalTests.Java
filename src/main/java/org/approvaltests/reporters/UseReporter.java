package org.approvaltests.reporters;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

import org.approvaltests.core.ApprovalFailureReporter;

@Retention(RetentionPolicy.RUNTIME)
public @interface UseReporter {
  Class<? extends ApprovalFailureReporter>[] value();
}
