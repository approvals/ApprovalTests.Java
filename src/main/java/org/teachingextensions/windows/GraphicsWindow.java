package org.teachingextensions.windows;

public class GraphicsWindow
{
  public static void show()
  {
    //    DisplayWindow displayWindow = new DisplayWindow();
    //    DisplayWindow.show();
  }
}
