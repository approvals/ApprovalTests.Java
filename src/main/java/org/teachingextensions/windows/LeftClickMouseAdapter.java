package org.teachingextensions.windows;

import java.awt.event.MouseEvent;
import java.awt.event.MouseListener;

import javax.swing.SwingUtilities;

public class LeftClickMouseAdapter implements MouseListener
{
  private final MouseLeftClickListener listener;
  public LeftClickMouseAdapter(MouseLeftClickListener listener)
  {
    this.listener = listener;
  }
  @Override
  public void mouseClicked(MouseEvent e)
  {
  }
  @Override
  public void mouseEntered(MouseEvent e)
  {
  }
  @Override
  public void mouseExited(MouseEvent e)
  {
  }
  @Override
  public void mousePressed(MouseEvent e)
  {
  }
  @Override
  public void mouseReleased(MouseEvent e)
  {
    if (SwingUtilities.isLeftMouseButton(e))
    {
      listener.onLeftMouseClick(e.getX(), e.getY());
    }
  }
}
