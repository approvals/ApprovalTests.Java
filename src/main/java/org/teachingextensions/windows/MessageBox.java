package org.teachingextensions.windows;

import javax.swing.JOptionPane;

import com.spun.util.NumberUtils;

/**
 * 
 * MessageBox is a window that can collect a numerical input from the user
 * or prints a message for the user to read. <br/>
 * 
 * 
 * 
 */
public class MessageBox
{
  private static MessageBoxInstance messageBox = new MessageBoxInstance();
  /**
   * Prints a request for a numerical input to the window. <br/>
   * <b>Example:</b> {@code  int cookies = MessageBox.askForNumericalInput("How many cookies would you like?");}
   * 
   * @param message
   *          the text to be displayed
   * @return the user input
   */
  public static int askForNumericalInput(String message)
  {
    return messageBox.askForNumericalInput(message);
  }
  /**
   * Prints a request for a text input to the window. <br/>
   * <b>Example:</b> {@code  String name = MessageBox.askForTextInput("What is your nickname?");}
   * 
   * @param message
   *          the text to be displayed
   * @return the user input
   */
  public static String askForTextInput(String message)
  {
    return messageBox.askForTextInput(message);
  }
  /**
   * Prints the message to the window. <br/>
   * <b>Example:</b> {@code  MessageBox.showMessage("Girl programmers rule!");}
   * 
   * @param message
   *          the text to be displayed
   */
  public static void showMessage(String message)
  {
    messageBox.showMessage(message);
  }
  public static void mock(MessageBoxInstance messageBoxMock)
  {
    messageBox = messageBoxMock;
  }
  public static class MessageBoxInstance
  {
    public int askForNumericalInput(String message)
    {
      String input = askForTextInput(message);
      return NumberUtils.load(input, 0);
    }
    public String askForTextInput(String message)
    {
      return JOptionPane.showInputDialog(message);
    }
    public void showMessage(String message)
    {
      JOptionPane.showMessageDialog(null, message);
    }
  }
}
