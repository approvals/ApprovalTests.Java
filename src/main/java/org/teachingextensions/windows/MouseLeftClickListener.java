package org.teachingextensions.windows;

public interface MouseLeftClickListener
{
  void onLeftMouseClick(int x, int y);
}
