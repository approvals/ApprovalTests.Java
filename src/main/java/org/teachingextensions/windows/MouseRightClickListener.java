package org.teachingextensions.windows;

public interface MouseRightClickListener
{
  void onRightMouseClick(int x, int y);
}
