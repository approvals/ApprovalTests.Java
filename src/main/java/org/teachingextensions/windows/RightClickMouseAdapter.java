package org.teachingextensions.windows;

import java.awt.event.MouseEvent;
import java.awt.event.MouseListener;

import javax.swing.SwingUtilities;

public class RightClickMouseAdapter implements MouseListener
{
  private final MouseRightClickListener listener;
  public RightClickMouseAdapter(MouseRightClickListener listener)
  {
    this.listener = listener;
  }
  @Override
  public void mouseClicked(MouseEvent e)
  {
  }
  @Override
  public void mouseEntered(MouseEvent e)
  {
  }
  @Override
  public void mouseExited(MouseEvent e)
  {
  }
  @Override
  public void mousePressed(MouseEvent e)
  {
  }
  @Override
  public void mouseReleased(MouseEvent e)
  {
    if (SwingUtilities.isRightMouseButton(e))
    {
      listener.onRightMouseClick(e.getX(), e.getY());
    }
  }
}
