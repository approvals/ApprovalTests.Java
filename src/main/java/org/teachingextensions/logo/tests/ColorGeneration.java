package org.teachingextensions.logo.tests;

import java.awt.Color;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collections;
import java.util.Comparator;
import java.util.HashMap;
import java.util.List;
import java.util.Map.Entry;

import junit.framework.TestCase;

import org.approvaltests.Approvals;
import org.approvaltests.reporters.UseReporter;
import org.approvaltests.reporters.windows.TortoiseTextDiffReporter;

import com.spun.util.StringUtils;
import com.spun.util.Tuple;
import com.spun.util.io.FileUtils;
import com.spun.util.tests.TestUtils;
import com.spun.util.velocity.ContextAware.ContextAwareMap;
import com.spun.util.velocity.VelocityParser;

@UseReporter(TortoiseTextDiffReporter.class)
public class ColorGeneration extends TestCase
{
  public void testGeneration() throws Exception
  {
    HashMap<String, List<Tuple<String, String>>> loadColors = loadColors();
    Approvals.verify(generateColors(loadColors, "colors.java.template"));
  }
  public void testHtmlDisplay() throws Exception
  {
    HashMap<String, List<Tuple<String, String>>> loadColors = loadColors();
    Approvals.verifyHtml(generateColors(loadColors, "colors.html"));
  }
  public void ptestSortingDisplay() throws Exception
  {
    HashMap<String, List<Tuple<String, String>>> loadColors = loadColors();
    TestUtils.displayText(generateColors(loadColors, "colors.sorted.txt"));
  }
  private HashMap<String, List<Tuple<String, String>>> loadColors()
  {
    HashMap<String, List<Tuple<String, String>>> colors = new HashMap<String, List<Tuple<String, String>>>();
    String[] split = FileUtils.readFromClassPath(getClass(), "colors.txt").split("\n");
    String currentColor = "";
    for (String line : split)
    {
      String[] parts = StringUtils.stripWhiteSpace(line).split(" ");
      //System.out.println(line);
      if (parts.length == 1)
      {
        currentColor = parts[0];
      }
      else
      {
        add(colors, currentColor, parts[0], parts[1]);
      }
    }
    return colors;
  }
  private String generateColors(HashMap<String, List<Tuple<String, String>>> colors, String template)
  {
    ContextAwareMap aware = new ContextAwareMap("colors", colors);
    aware.put("finder", this);
    Object[] keys = colors.keySet().toArray();
    Arrays.sort(keys);
    aware.put("keys", keys);
    return VelocityParser.parseFromClassPath(this.getClass(), template, aware);
  }
  public static String getOtherColors(HashMap<String, List<Tuple<String, String>>> colors, String key, String color)
  {
    String out = "";
    for (Entry<String, List<Tuple<String, String>>> entry : colors.entrySet())
    {
      if (!entry.getKey().equals(key))
      {
        if (containsColor(entry.getValue(), color))
        {
          out += entry.getKey() + ", ";
        }
      }
    }
    return out;
  }
  private static boolean containsColor(List<Tuple<String, String>> value, String color)
  {
    for (Tuple<String, String> tuple : value)
    {
      if (color.equals(tuple.getFirst())) { return true; }
    }
    return false;
  }
  private void add(HashMap<String, List<Tuple<String, String>>> colors, String colorGroup, String name,
      String hexValue)
  {
    List<Tuple<String, String>> list = colors.get(colorGroup);
    if (list == null)
    {
      list = new ArrayList<Tuple<String, String>>();
      colors.put(colorGroup, list);
    }
    list.add(new Tuple<String, String>(name, hexValue));
    Collections.sort(list, new BrightnessComparator());
  }
  public class BrightnessComparator implements Comparator<Tuple<String, String>>
  {
    @Override
    public int compare(Tuple<String, String> o1, Tuple<String, String> o2)
    {
      Color c1 = Color.decode(o1.getSecond());
      Color c2 = Color.decode(o2.getSecond());
      Integer b1 = c1.getGreen() + c1.getRed() + c1.getBlue();
      Integer b2 = c2.getGreen() + c2.getRed() + c2.getBlue();
      return b1.compareTo(b2);
    }
  }
}
