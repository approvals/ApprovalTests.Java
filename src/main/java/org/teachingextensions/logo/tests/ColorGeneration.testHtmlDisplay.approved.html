<table border="1">
	<tr>
		<th>Blues</th>
	</tr>
		<tr>
		<td bgcolor="#000080">Navy</td>
		<td bgcolor="#000080"></td>
	</tr>
		<tr>
		<td bgcolor="#00008B">DarkBlue</td>
		<td bgcolor="#00008B"></td>
	</tr>
		<tr>
		<td bgcolor="#191970">MidnightBlue</td>
		<td bgcolor="#191970"></td>
	</tr>
		<tr>
		<td bgcolor="#0000CD">MediumBlue</td>
		<td bgcolor="#0000CD"></td>
	</tr>
		<tr>
		<td bgcolor="#0000FF">Blue</td>
		<td bgcolor="#0000FF"></td>
	</tr>
		<tr>
		<td bgcolor="#008080">Teal</td>
		<td bgcolor="#008080">Greens, </td>
	</tr>
		<tr>
		<td bgcolor="#483D8B">DarkSlateBlue</td>
		<td bgcolor="#483D8B">Purples, </td>
	</tr>
		<tr>
		<td bgcolor="#008B8B">DarkCyan</td>
		<td bgcolor="#008B8B">Greens, </td>
	</tr>
		<tr>
		<td bgcolor="#4682B4">SteelBlue</td>
		<td bgcolor="#4682B4"></td>
	</tr>
		<tr>
		<td bgcolor="#4169E1">RoyalBlue</td>
		<td bgcolor="#4169E1"></td>
	</tr>
		<tr>
		<td bgcolor="#6A5ACD">SlateBlue</td>
		<td bgcolor="#6A5ACD">Purples, </td>
	</tr>
		<tr>
		<td bgcolor="#5F9EA0">CadetBlue</td>
		<td bgcolor="#5F9EA0"></td>
	</tr>
		<tr>
		<td bgcolor="#00CED1">DarkTurquoise</td>
		<td bgcolor="#00CED1">Greens, </td>
	</tr>
		<tr>
		<td bgcolor="#1E90FF">DodgerBlue</td>
		<td bgcolor="#1E90FF"></td>
	</tr>
		<tr>
		<td bgcolor="#00BFFF">DeepSkyBlue</td>
		<td bgcolor="#00BFFF"></td>
	</tr>
		<tr>
		<td bgcolor="#7B68EE">MediumSlateBlue</td>
		<td bgcolor="#7B68EE">Purples, </td>
	</tr>
		<tr>
		<td bgcolor="#48D1CC">MediumTurquoise</td>
		<td bgcolor="#48D1CC">Greens, </td>
	</tr>
		<tr>
		<td bgcolor="#6495ED">CornflowerBlue</td>
		<td bgcolor="#6495ED"></td>
	</tr>
		<tr>
		<td bgcolor="#40E0D0">Turquoise</td>
		<td bgcolor="#40E0D0">Greens, </td>
	</tr>
		<tr>
		<td bgcolor="#00FFFF">Aqua</td>
		<td bgcolor="#00FFFF"></td>
	</tr>
		<tr>
		<td bgcolor="#00FFFF">Cyan</td>
		<td bgcolor="#00FFFF"></td>
	</tr>
		<tr>
		<td bgcolor="#87CEEB">SkyBlue</td>
		<td bgcolor="#87CEEB"></td>
	</tr>
		<tr>
		<td bgcolor="#87CEFA">LightSkyBlue</td>
		<td bgcolor="#87CEFA"></td>
	</tr>
		<tr>
		<td bgcolor="#7FFFD4">Aquamarine</td>
		<td bgcolor="#7FFFD4">Greens, </td>
	</tr>
		<tr>
		<td bgcolor="#B0C4DE">LightSteelBlue</td>
		<td bgcolor="#B0C4DE"></td>
	</tr>
		<tr>
		<td bgcolor="#ADD8E6">LightBlue</td>
		<td bgcolor="#ADD8E6"></td>
	</tr>
		<tr>
		<td bgcolor="#B0E0E6">PowderBlue</td>
		<td bgcolor="#B0E0E6"></td>
	</tr>
		<tr>
		<td bgcolor="#AFEEEE">PaleTurquoise</td>
		<td bgcolor="#AFEEEE">Greens, </td>
	</tr>
		<tr>
		<td bgcolor="#E0FFFF">LightCyan</td>
		<td bgcolor="#E0FFFF"></td>
	</tr>
		<tr>
		<td bgcolor="#F0F8FF">AliceBlue</td>
		<td bgcolor="#F0F8FF">Whites, </td>
	</tr>
		<tr>
		<td bgcolor="#F0FFFF">Azure</td>
		<td bgcolor="#F0FFFF">Whites, </td>
	</tr>
	</table>
<table border="1">
	<tr>
		<th>Browns</th>
	</tr>
		<tr>
		<td bgcolor="#800000">Maroon</td>
		<td bgcolor="#800000"></td>
	</tr>
		<tr>
		<td bgcolor="#8B4513">SaddleBrown</td>
		<td bgcolor="#8B4513"></td>
	</tr>
		<tr>
		<td bgcolor="#A52A2A">Brown</td>
		<td bgcolor="#A52A2A"></td>
	</tr>
		<tr>
		<td bgcolor="#A0522D">Sienna</td>
		<td bgcolor="#A0522D"></td>
	</tr>
		<tr>
		<td bgcolor="#B8860B">DarkGoldenrod</td>
		<td bgcolor="#B8860B">Yellows, </td>
	</tr>
		<tr>
		<td bgcolor="#D2691E">Chocolate</td>
		<td bgcolor="#D2691E"></td>
	</tr>
		<tr>
		<td bgcolor="#CD853F">Peru</td>
		<td bgcolor="#CD853F"></td>
	</tr>
		<tr>
		<td bgcolor="#DAA520">Goldenrod</td>
		<td bgcolor="#DAA520">Yellows, </td>
	</tr>
		<tr>
		<td bgcolor="#BC8F8F">RosyBrown</td>
		<td bgcolor="#BC8F8F">Purples, </td>
	</tr>
		<tr>
		<td bgcolor="#F4A460">SandyBrown</td>
		<td bgcolor="#F4A460"></td>
	</tr>
		<tr>
		<td bgcolor="#D2B48C">Tan</td>
		<td bgcolor="#D2B48C"></td>
	</tr>
		<tr>
		<td bgcolor="#DEB887">BurlyWood</td>
		<td bgcolor="#DEB887"></td>
	</tr>
		<tr>
		<td bgcolor="#F5DEB3">Wheat</td>
		<td bgcolor="#F5DEB3"></td>
	</tr>
		<tr>
		<td bgcolor="#FFDEAD">NavajoWhite</td>
		<td bgcolor="#FFDEAD"></td>
	</tr>
		<tr>
		<td bgcolor="#FFE4C4">Bisque</td>
		<td bgcolor="#FFE4C4"></td>
	</tr>
		<tr>
		<td bgcolor="#FFEBCD">BlanchedAlmond</td>
		<td bgcolor="#FFEBCD"></td>
	</tr>
		<tr>
		<td bgcolor="#FFF8DC">Cornsilk</td>
		<td bgcolor="#FFF8DC"></td>
	</tr>
	</table>
<table border="1">
	<tr>
		<th>Grays</th>
	</tr>
		<tr>
		<td bgcolor="#000000">Black</td>
		<td bgcolor="#000000"></td>
	</tr>
		<tr>
		<td bgcolor="#2F4F4F">DarkSlateGray</td>
		<td bgcolor="#2F4F4F"></td>
	</tr>
		<tr>
		<td bgcolor="#696969">DimGray</td>
		<td bgcolor="#696969"></td>
	</tr>
		<tr>
		<td bgcolor="#808080">Gray</td>
		<td bgcolor="#808080"></td>
	</tr>
		<tr>
		<td bgcolor="#708090">SlateGray</td>
		<td bgcolor="#708090"></td>
	</tr>
		<tr>
		<td bgcolor="#778899">LightSlateGray</td>
		<td bgcolor="#778899"></td>
	</tr>
		<tr>
		<td bgcolor="#A9A9A9">DarkGray</td>
		<td bgcolor="#A9A9A9"></td>
	</tr>
		<tr>
		<td bgcolor="#C0C0C0">Silver</td>
		<td bgcolor="#C0C0C0"></td>
	</tr>
		<tr>
		<td bgcolor="#D3D3D3">LightGray</td>
		<td bgcolor="#D3D3D3"></td>
	</tr>
		<tr>
		<td bgcolor="#DCDCDC">Gainsboro</td>
		<td bgcolor="#DCDCDC"></td>
	</tr>
	</table>
<table border="1">
	<tr>
		<th>Greens</th>
	</tr>
		<tr>
		<td bgcolor="#006400">DarkGreen</td>
		<td bgcolor="#006400"></td>
	</tr>
		<tr>
		<td bgcolor="#008000">Green</td>
		<td bgcolor="#008000"></td>
	</tr>
		<tr>
		<td bgcolor="#228B22">ForestGreen</td>
		<td bgcolor="#228B22"></td>
	</tr>
		<tr>
		<td bgcolor="#556B2F">DarkOliveGreen</td>
		<td bgcolor="#556B2F"></td>
	</tr>
		<tr>
		<td bgcolor="#00FF00">Lime</td>
		<td bgcolor="#00FF00"></td>
	</tr>
		<tr>
		<td bgcolor="#808000">Olive</td>
		<td bgcolor="#808000"></td>
	</tr>
		<tr>
		<td bgcolor="#008080">Teal</td>
		<td bgcolor="#008080">Blues, </td>
	</tr>
		<tr>
		<td bgcolor="#2E8B57">SeaGreen</td>
		<td bgcolor="#2E8B57"></td>
	</tr>
		<tr>
		<td bgcolor="#008B8B">DarkCyan</td>
		<td bgcolor="#008B8B">Blues, </td>
	</tr>
		<tr>
		<td bgcolor="#6B8E23">OliveDrab</td>
		<td bgcolor="#6B8E23"></td>
	</tr>
		<tr>
		<td bgcolor="#32CD32">LimeGreen</td>
		<td bgcolor="#32CD32"></td>
	</tr>
		<tr>
		<td bgcolor="#3CB371">MediumSeaGreen</td>
		<td bgcolor="#3CB371"></td>
	</tr>
		<tr>
		<td bgcolor="#7CFC00">LawnGreen</td>
		<td bgcolor="#7CFC00"></td>
	</tr>
		<tr>
		<td bgcolor="#20B2AA">LightSeaGreen</td>
		<td bgcolor="#20B2AA"></td>
	</tr>
		<tr>
		<td bgcolor="#7FFF00">Chartreuse</td>
		<td bgcolor="#7FFF00"></td>
	</tr>
		<tr>
		<td bgcolor="#00FF7F">SpringGreen</td>
		<td bgcolor="#00FF7F"></td>
	</tr>
		<tr>
		<td bgcolor="#00FA9A">MediumSpringGreen</td>
		<td bgcolor="#00FA9A"></td>
	</tr>
		<tr>
		<td bgcolor="#9ACD32">YellowGreen</td>
		<td bgcolor="#9ACD32"></td>
	</tr>
		<tr>
		<td bgcolor="#00CED1">DarkTurquoise</td>
		<td bgcolor="#00CED1">Blues, </td>
	</tr>
		<tr>
		<td bgcolor="#8FBC8F">DarkSeaGreen</td>
		<td bgcolor="#8FBC8F"></td>
	</tr>
		<tr>
		<td bgcolor="#ADFF2F">GreenYellow</td>
		<td bgcolor="#ADFF2F"></td>
	</tr>
		<tr>
		<td bgcolor="#66CDAA">MediumAquamarine</td>
		<td bgcolor="#66CDAA"></td>
	</tr>
		<tr>
		<td bgcolor="#48D1CC">MediumTurquoise</td>
		<td bgcolor="#48D1CC">Blues, </td>
	</tr>
		<tr>
		<td bgcolor="#40E0D0">Turquoise</td>
		<td bgcolor="#40E0D0">Blues, </td>
	</tr>
		<tr>
		<td bgcolor="#90EE90">LightGreen</td>
		<td bgcolor="#90EE90"></td>
	</tr>
		<tr>
		<td bgcolor="#98FB98">PaleGreen</td>
		<td bgcolor="#98FB98"></td>
	</tr>
		<tr>
		<td bgcolor="#7FFFD4">Aquamarine</td>
		<td bgcolor="#7FFFD4">Blues, </td>
	</tr>
		<tr>
		<td bgcolor="#AFEEEE">PaleTurquoise</td>
		<td bgcolor="#AFEEEE">Blues, </td>
	</tr>
	</table>
<table border="1">
	<tr>
		<th>Oranges</th>
	</tr>
		<tr>
		<td bgcolor="#FF4500">OrangeRed</td>
		<td bgcolor="#FF4500">Reds, </td>
	</tr>
		<tr>
		<td bgcolor="#FF8C00">DarkOrange</td>
		<td bgcolor="#FF8C00"></td>
	</tr>
		<tr>
		<td bgcolor="#FFA500">Orange</td>
		<td bgcolor="#FFA500"></td>
	</tr>
		<tr>
		<td bgcolor="#FF6347">Tomato</td>
		<td bgcolor="#FF6347">Reds, </td>
	</tr>
		<tr>
		<td bgcolor="#FF7F50">Coral</td>
		<td bgcolor="#FF7F50"></td>
	</tr>
		<tr>
		<td bgcolor="#FFA07A">LightSalmon</td>
		<td bgcolor="#FFA07A">Reds, </td>
	</tr>
	</table>
<table border="1">
	<tr>
		<th>Pinks</th>
	</tr>
		<tr>
		<td bgcolor="#C71585">MediumVioletRed</td>
		<td bgcolor="#C71585">Reds, Purples, </td>
	</tr>
		<tr>
		<td bgcolor="#FF1493">DeepPink</td>
		<td bgcolor="#FF1493"></td>
	</tr>
		<tr>
		<td bgcolor="#DB7093">PaleVioletRed</td>
		<td bgcolor="#DB7093">Reds, Purples, </td>
	</tr>
		<tr>
		<td bgcolor="#FF00FF">Fuchsia</td>
		<td bgcolor="#FF00FF">Purples, </td>
	</tr>
		<tr>
		<td bgcolor="#FF69B4">HotPink</td>
		<td bgcolor="#FF69B4"></td>
	</tr>
		<tr>
		<td bgcolor="#FFB6C1">LightPink</td>
		<td bgcolor="#FFB6C1"></td>
	</tr>
		<tr>
		<td bgcolor="#FFC0CB">Pink</td>
		<td bgcolor="#FFC0CB"></td>
	</tr>
	</table>
<table border="1">
	<tr>
		<th>Purples</th>
	</tr>
		<tr>
		<td bgcolor="#4B0082">Indigo</td>
		<td bgcolor="#4B0082"></td>
	</tr>
		<tr>
		<td bgcolor="#800080">Purple</td>
		<td bgcolor="#800080"></td>
	</tr>
		<tr>
		<td bgcolor="#483D8B">DarkSlateBlue</td>
		<td bgcolor="#483D8B">Blues, </td>
	</tr>
		<tr>
		<td bgcolor="#8B008B">DarkMagenta</td>
		<td bgcolor="#8B008B"></td>
	</tr>
		<tr>
		<td bgcolor="#C71585">MediumVioletRed</td>
		<td bgcolor="#C71585">Reds, Pinks, </td>
	</tr>
		<tr>
		<td bgcolor="#9400D3">DarkViolet</td>
		<td bgcolor="#9400D3"></td>
	</tr>
		<tr>
		<td bgcolor="#6A5ACD">SlateBlue</td>
		<td bgcolor="#6A5ACD">Blues, </td>
	</tr>
		<tr>
		<td bgcolor="#8A2BE2">BlueViolet</td>
		<td bgcolor="#8A2BE2"></td>
	</tr>
		<tr>
		<td bgcolor="#9932CC">DarkOrchid</td>
		<td bgcolor="#9932CC"></td>
	</tr>
		<tr>
		<td bgcolor="#7B68EE">MediumSlateBlue</td>
		<td bgcolor="#7B68EE">Blues, </td>
	</tr>
		<tr>
		<td bgcolor="#BC8F8F">RosyBrown</td>
		<td bgcolor="#BC8F8F">Browns, </td>
	</tr>
		<tr>
		<td bgcolor="#9370DB">MediumPurple</td>
		<td bgcolor="#9370DB"></td>
	</tr>
		<tr>
		<td bgcolor="#DB7093">PaleVioletRed</td>
		<td bgcolor="#DB7093">Reds, Pinks, </td>
	</tr>
		<tr>
		<td bgcolor="#BA55D3">MediumOrchid</td>
		<td bgcolor="#BA55D3"></td>
	</tr>
		<tr>
		<td bgcolor="#FF00FF">Fuchsia</td>
		<td bgcolor="#FF00FF">Pinks, </td>
	</tr>
		<tr>
		<td bgcolor="#FF00FF">Magenta</td>
		<td bgcolor="#FF00FF"></td>
	</tr>
		<tr>
		<td bgcolor="#DA70D6">Orchid</td>
		<td bgcolor="#DA70D6"></td>
	</tr>
		<tr>
		<td bgcolor="#DDA0DD">Plum</td>
		<td bgcolor="#DDA0DD"></td>
	</tr>
		<tr>
		<td bgcolor="#EE82EE">Violet</td>
		<td bgcolor="#EE82EE"></td>
	</tr>
		<tr>
		<td bgcolor="#D8BFD8">Thistle</td>
		<td bgcolor="#D8BFD8"></td>
	</tr>
		<tr>
		<td bgcolor="#E6E6FA">Lavender</td>
		<td bgcolor="#E6E6FA"></td>
	</tr>
	</table>
<table border="1">
	<tr>
		<th>Reds</th>
	</tr>
		<tr>
		<td bgcolor="#8B0000">DarkRed</td>
		<td bgcolor="#8B0000"></td>
	</tr>
		<tr>
		<td bgcolor="#B22222">FireBrick</td>
		<td bgcolor="#B22222"></td>
	</tr>
		<tr>
		<td bgcolor="#FF0000">Red</td>
		<td bgcolor="#FF0000"></td>
	</tr>
		<tr>
		<td bgcolor="#DC143C">Crimson</td>
		<td bgcolor="#DC143C"></td>
	</tr>
		<tr>
		<td bgcolor="#FF4500">OrangeRed</td>
		<td bgcolor="#FF4500">Oranges, </td>
	</tr>
		<tr>
		<td bgcolor="#C71585">MediumVioletRed</td>
		<td bgcolor="#C71585">Pinks, Purples, </td>
	</tr>
		<tr>
		<td bgcolor="#CD5C5C">IndianRed</td>
		<td bgcolor="#CD5C5C"></td>
	</tr>
		<tr>
		<td bgcolor="#FF6347">Tomato</td>
		<td bgcolor="#FF6347">Oranges, </td>
	</tr>
		<tr>
		<td bgcolor="#DB7093">PaleVioletRed</td>
		<td bgcolor="#DB7093">Pinks, Purples, </td>
	</tr>
		<tr>
		<td bgcolor="#FA8072">Salmon</td>
		<td bgcolor="#FA8072"></td>
	</tr>
		<tr>
		<td bgcolor="#F08080">LightCoral</td>
		<td bgcolor="#F08080"></td>
	</tr>
		<tr>
		<td bgcolor="#E9967A">DarkSalmon</td>
		<td bgcolor="#E9967A"></td>
	</tr>
		<tr>
		<td bgcolor="#FFA07A">LightSalmon</td>
		<td bgcolor="#FFA07A">Oranges, </td>
	</tr>
	</table>
<table border="1">
	<tr>
		<th>Whites</th>
	</tr>
		<tr>
		<td bgcolor="#FAEBD7">AntiqueWhite</td>
		<td bgcolor="#FAEBD7"></td>
	</tr>
		<tr>
		<td bgcolor="#FFE4E1">MistyRose</td>
		<td bgcolor="#FFE4E1"></td>
	</tr>
		<tr>
		<td bgcolor="#F5F5DC">Beige</td>
		<td bgcolor="#F5F5DC"></td>
	</tr>
		<tr>
		<td bgcolor="#FAF0E6">Linen</td>
		<td bgcolor="#FAF0E6"></td>
	</tr>
		<tr>
		<td bgcolor="#FDF5E6">OldLace</td>
		<td bgcolor="#FDF5E6"></td>
	</tr>
		<tr>
		<td bgcolor="#F0FFF0">Honeydew</td>
		<td bgcolor="#F0FFF0"></td>
	</tr>
		<tr>
		<td bgcolor="#F5F5F5">WhiteSmoke</td>
		<td bgcolor="#F5F5F5"></td>
	</tr>
		<tr>
		<td bgcolor="#FFF5EE">Seashell</td>
		<td bgcolor="#FFF5EE"></td>
	</tr>
		<tr>
		<td bgcolor="#FFF0F5">LavenderBlush</td>
		<td bgcolor="#FFF0F5"></td>
	</tr>
		<tr>
		<td bgcolor="#F0F8FF">AliceBlue</td>
		<td bgcolor="#F0F8FF">Blues, </td>
	</tr>
		<tr>
		<td bgcolor="#FFFAF0">FloralWhite</td>
		<td bgcolor="#FFFAF0"></td>
	</tr>
		<tr>
		<td bgcolor="#F5FFFA">MintCream</td>
		<td bgcolor="#F5FFFA"></td>
	</tr>
		<tr>
		<td bgcolor="#F0FFFF">Azure</td>
		<td bgcolor="#F0FFFF">Blues, </td>
	</tr>
		<tr>
		<td bgcolor="#FFFFF0">Ivory</td>
		<td bgcolor="#FFFFF0"></td>
	</tr>
		<tr>
		<td bgcolor="#F8F8FF">GhostWhite</td>
		<td bgcolor="#F8F8FF"></td>
	</tr>
		<tr>
		<td bgcolor="#FFFAFA">Snow</td>
		<td bgcolor="#FFFAFA"></td>
	</tr>
		<tr>
		<td bgcolor="#FFFFFF">White</td>
		<td bgcolor="#FFFFFF"></td>
	</tr>
	</table>
<table border="1">
	<tr>
		<th>Yellows</th>
	</tr>
		<tr>
		<td bgcolor="#B8860B">DarkGoldenrod</td>
		<td bgcolor="#B8860B">Browns, </td>
	</tr>
		<tr>
		<td bgcolor="#DAA520">Goldenrod</td>
		<td bgcolor="#DAA520">Browns, </td>
	</tr>
		<tr>
		<td bgcolor="#FFD700">Gold</td>
		<td bgcolor="#FFD700"></td>
	</tr>
		<tr>
		<td bgcolor="#BDB76B">DarkKhaki</td>
		<td bgcolor="#BDB76B"></td>
	</tr>
		<tr>
		<td bgcolor="#FFFF00">Yellow</td>
		<td bgcolor="#FFFF00"></td>
	</tr>
		<tr>
		<td bgcolor="#F0E68C">Khaki</td>
		<td bgcolor="#F0E68C"></td>
	</tr>
		<tr>
		<td bgcolor="#EEE8AA">PaleGoldenrod</td>
		<td bgcolor="#EEE8AA"></td>
	</tr>
		<tr>
		<td bgcolor="#FFDAB9">PeachPuff</td>
		<td bgcolor="#FFDAB9"></td>
	</tr>
		<tr>
		<td bgcolor="#FFE4B5">Moccasin</td>
		<td bgcolor="#FFE4B5"></td>
	</tr>
		<tr>
		<td bgcolor="#FFEFD5">PapayaWhip</td>
		<td bgcolor="#FFEFD5"></td>
	</tr>
		<tr>
		<td bgcolor="#FFFACD">LemonChiffon</td>
		<td bgcolor="#FFFACD"></td>
	</tr>
		<tr>
		<td bgcolor="#FAFAD2">LightGoldenrodYellow</td>
		<td bgcolor="#FAFAD2"></td>
	</tr>
		<tr>
		<td bgcolor="#FFFFE0">LightYellow</td>
		<td bgcolor="#FFFFE0"></td>
	</tr>
	</table>
